### PR TITLE
[export] Preserve requires_grad for export inputs.

### PR DESCRIPTION
**Pull request by zhxchen17**
Summary: Today meta['val'] on placeholder nodes doesn't preserve the consistent requires_grad information with the original inputs. Seems there's no easy way to fix this directly at proxy tensor layer. This is useful for reexporting joint graph.

Test Plan: test_preserve_requires_grad_placeholders

Differential Revision: D58555651

### CODE REVIEW

File: test/export/test_export.py [hunk 1]
```diff
@@ -5382,6 +5382,25 @@ def forward(self, w, x, y, z):
                 _disable_forced_specializations=True,
             )
 
+    @testing.expectedFailureSerDer
```
Comment (tugsbayasgalan) [inline test/export/test_export.py:5385]: What fails here? Could you file issue?

File: torch/export/_trace.py [hunk 2]
```diff
@@ -339,6 +339,41 @@ def _get_param_buffer_mapping(
...
+    placeholders = [node for node in gm.graph.nodes if node.op == "placeholder"]
+    assert len(sig.input_specs) == len(placeholders)
+    i = 0
+    for node, spec in zip(placeholders, sig.input_specs):
+        if spec.kind in (
+            InputKind.PARAMETER,
+            InputKind.BUFFER,
+        ):
+            assert spec.target is not None
+            node.meta["val"].requires_grad = fake_params_buffers[
```
Comment (tugsbayasgalan) [inline torch/export/_trace.py:358]: Should we do this at fake tensor constructor? I think fake tensor should reflect faithfully what the original tensor does.

Comment (zhxchen17) [reply torch/export/_trace.py:358]: we couldn't right now. see comments here: https://fburl.com/code/vgl33i42

File: test/export/test_export.py (hunk 1 above)
Comment (zhxchen17) [reply test/export/test_export.py:5385]: I think requires_grad are not preserved with roundtripping. I can file an issue later.

File: torch/export/_trace.py (hunk 2 above)
Comment (tugsbayasgalan) [reply torch/export/_trace.py:358]: Do we not use the same inputs for "val" field? Interesting...

File: test/export/test_export.py (hunk 1 above)
Comment (tugsbayasgalan) [reply test/export/test_export.py:5385]: Can you leave a message describing what doesn't work then?

**Review by tugsbayasgalan — APPROVED** [review]

**Pull request closed without merging**